R/isovist.R

```diff
@@ -5,9 +5,9 @@
 #'
 #' @return object of class sfc_POINT
 #' @export
-get_viewpoints <- function(x, density = 1/50) {
+get_viewpoints <- function(x, density = 1 / 50) {
   if (density <= 0) stop("Density must be a non-zero positive number")
-  if (sf::st_geometry_type(x) %in% c("POINT", "MULTIPOINT")) {
+  if (any(sf::st_geometry_type(x) %in% c("POINT", "MULTIPOINT"))) {
```

Comment:
Is `any()` needed in case the input is a `GEOMETRYCOLLECTION` that includes a `POINT` or `MULTIPOINT`? Otherwise, `%in%` already returns a single logical value.

Reply:
It is needed any time `x` contains multiple features. For N features,  `sf::st_geometry_type(x) %in% c("POINT", "MULTIPOINT")` is a vector with length N, so we need to use `any` to make sure none of the features are of type point or multipoint

Reply:
You're right. Initially, this function was used on a single point (although did not check for this), but in the context of vectorised isovist analysis, this should work with multiple points as well.